### PR TITLE
Prevent accounts without account numbers from breaking the payments page

### PR DESCRIPTION
**Pull request by stitesExpensify**
### Details
1. Do not return un-issued or deleted cards
2. If we get an account without an account number just don't display the description.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/171546

### Tests
1. Log into e.cash with an account that has both a bank account and credit card
2. Make sure there are descriptions under each with the last 4 of the account numbers

### QA Steps
1. Ping @joekaufmanexpensify and ask him to go to the payments page
2. His page should not crash

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

File: src/pages/settings/Payments/PaymentMethodList.js [hunk 1]
```diff
@@ -83,12 +83,17 @@ class PaymentMethodList extends Component {
         _.each(this.props.bankAccountList, (bankAccount) => {
             // Add all bank accounts besides the wallet
             if (bankAccount.type !== CONST.BANK_ACCOUNT_TYPES.WALLET) {
+                const formattedBankAccountNumber = bankAccount.accountNumber
```
Comment (Dal-Papa) [inline src/pages/settings/Payments/PaymentMethodList.js:86]: The famous line broken down 😂

**Review by Dal-Papa — APPROVED** [review]
Any chance you (or @joekaufmanexpensify) can provide a screenshot ?

**Comment by joekaufmanexpensify** [comment]
@Dal-Papa Are you looking for a screenshot of how my screen looks when I try to access the payments page from my test account?

**Comment by Dal-Papa** [comment]
Ideally a screenshot of your screen with the fix that @stitesExpensify provided, but I think we can simply QA when that's out.

**Comment by joekaufmanexpensify** [comment]
Cool sounds good, feel free to ping me when it's time to QA!

**Comment by stitesExpensify** [comment]
Account without account number and no crash

**Comment by isagoico** [comment]
@joekaufmanexpensify Can you go to your payments page and check if the app crashes? 😬 (following PR orders)

**Comment by joekaufmanexpensify** [comment]
I was able to access the Payments page from my expensifail account on staging!

**Comment by stitesExpensify** [comment]
:tada: Thanks @joekaufmanexpensify

**Comment by joekaufmanexpensify** [comment]
you're welcome!

**Comment by stitesExpensify** [comment]
This issue can be checked off, but it looks like something weird is still going on though, we shouldn't be putting non-activated cards in the wallet since they can't be used. Would you mind making an issue for that @joekaufmanexpensify ?

**Comment by joekaufmanexpensify** [comment]
Wouldn't we want that listed there so the user can click the validate that card in the future? Even though I'm not using this card because it was sent by mistake, it's still a card that exists. Or is it just not expected behavior that the card is listed there right now because validating expensify cards in newdot doesn't exist yet?

**Comment by stitesExpensify** [comment]
Yeah my thinking was that we would not show it for now since it is not usable, and then we'll start showing those again once we decide we want the ability to activate in newDot

**Comment by joekaufmanexpensify** [comment]
We are supposed to show the card in NewDot once it is activated though right? If that is what we do, then it could make sense to just add `Not Activated` or something similar now rather remove it. Because then we'd just need to add it back later.

**Comment by stitesExpensify** [comment]
Well the cards are in different states. So basically the front end would only get activated cards from the server which means that when it is activated, it will automatically show up

**Comment by joekaufmanexpensify** [comment]
Ah got it, so we haven't really considered how the activation flow will work for Expensify Cards in NewDot at all? And we haven't even mapped out how we want un-activated cards to appear in NewDot yet for N6, so we should remove now and then add something like this when we've actively designed how this will work?

**Comment by stitesExpensify** [comment]
Yep! That's what I'm thinking :)

**Comment by joekaufmanexpensify** [comment]
Sounds good. I'll make a new issue today!

**Pull request merged**